aws-lc-rs/src/key_wrap/tests.rs

```diff
@@ -0,0 +1,689 @@
+        0xda, 0xe0, 0xf7, 0x62, 0xb0, 0x77, 0xb1, 0x24, 0x4b, 0x44, 0x31, 0x47, 0xf5, 0x69, 0x57,
+        0x99, 0xfa
+    ],
+    &[
+        0x77, 0x6b, 0x1e, 0x91, 0xe9, 0x35, 0xd1, 0xf8, 0x0a, 0x53, 0x79, 0x02, 0x18, 0x6d, 0x6b,
+        0x00, 0xdf, 0xc6, 0xaf, 0xc1, 0x20, 0x00, 0xf1, 0xbd, 0xe9, 0x13, 0xdf, 0x5d, 0x67, 0x40,
+        0x70, 0x61, 0xdb, 0x82, 0x27, 0xfc, 0xd0, 0x89, 0x53, 0xd4
+    ],
+    &[
+        0xe6, 0x17, 0x83, 0x1c, 0x7d, 0xb8, 0x03, 0x8f, 0xda, 0x4c, 0x59, 0x40, 0x37, 0x75, 0xc3,
+        0xd4, 0x35, 0x13, 0x6a, 0x56, 0x6f, 0x35, 0x09, 0xc2, 0x73, 0xe1, 0xda, 0x1e, 0xf9, 0xf5,
+        0x0a, 0xea
+    ]
+);
+
+nist_aes_key_unwrap_test!(
+    kw_ad_aes256_256bit_len_fail,
+    &AES_256,
+    &[
+        0x3c, 0x7c, 0x55, 0x9f, 0xb9, 0x9d, 0x2e, 0x3f, 0x82, 0x80, 0xc9, 0xbe, 0x14, 0xb0, 0xf7,
+        0xb6, 0x76, 0xa3, 0x20, 0x53, 0xeb, 0xa8, 0xf7, 0xaf, 0xbb, 0x43, 0x04, 0xc1, 0x17, 0xa6,
+        0x50, 0x69
+    ],
+    &[
+        0x86, 0x1b, 0x0a, 0x15, 0xbf, 0x59, 0x07, 0xb5, 0x51, 0xbc, 0x94, 0x82, 0xbc, 0x4d, 0xe3,
+        0x61, 0xde, 0x64, 0x5f, 0x18, 0xf9, 0x7f, 0xd8, 0x0f, 0xff, 0xa5, 0xb9, 0x68, 0x79, 0x23,
+        0x82, 0x59, 0xc6, 0x67, 0x7e, 0xb5, 0x05, 0x96, 0x20, 0x5b
+    ]
+);
+
+macro_rules! wrap_input_output_invalid_test {
+    ($name:ident, $input_len:expr, $output_len:expr) => {
+        #[test]
+        fn $name() {
+            let kek = AesKek::new(&AES_128, &[16u8; 16]).expect("key creation successful");
+
+            let input_len: usize = $input_len.try_into().unwrap();
+            let output_len: usize = $output_len.try_into().unwrap();
+
+            let input = vec![42u8; input_len];
+            let mut output = vec![0u8; output_len];
+
+            kek.wrap(input.as_slice(), output.as_mut_slice())
+                .expect_err("failure");
+        }
+    };
+}
+
+// Input length < 16
+wrap_input_output_invalid_test!(wrap_input_len_less_than_min, 15, 23);
+
+// Input length % 8 != 0
+wrap_input_output_invalid_test!(wrap_input_len_not_multiple_of_eight, 17, 25);
+
+// Output length < Input length - 8
+wrap_input_output_invalid_test!(wrap_output_len_too_small, 16, 8);
+
+// Input Length == 0
+wrap_input_output_invalid_test!(wrap_padded_input_len_zero, 0, 8);
+
+macro_rules! unwrap_input_output_invalid_test {
+    ($name:ident, $input_len:expr, $output_len:expr) => {
+        #[test]
+        fn $name() {
+            let kek = AesKek::new(&AES_128, &[16u8; 16]).expect("key creation successful");
+
+            let input_len: usize = $input_len.try_into().unwrap();
+            let output_len: usize = $output_len.try_into().unwrap();
+
+            let input = vec![42u8; input_len];
+            let mut output = vec![0u8; output_len];
+
+            kek.unwrap(input.as_slice(), output.as_mut_slice())
+                .expect_err("failure");
+        }
+    };
+}
+
+macro_rules! unwrap_with_padding_input_output_invalid_test {
```

Comment:
there's `wrap_input_output_invalid_test` and `unwrap_input_output_invalid_test` for KW, but only `unwrap_with_padding_input_output_invalid_test` for KWP. is that asymmetry intentional?

Reply:
Probably a miss will take another look.